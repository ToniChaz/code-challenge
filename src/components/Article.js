import React, { Component } from 'react'
import { Col, CardLink, Card, CardText, CardBody, CardTitle, CardFooter } from 'reactstrap'

class Article extends Component {
  render() {
    const { article: { author, title, excerpt } } = this.props

    return (
      <Col xs="12" sm="6" md="4">
        <Card>
          <CardBody>
            <CardTitle>{title}</CardTitle>
            <CardText>{excerpt.substring(0, 100)}</CardText>
            <CardLink href="#">Read more</CardLink>
          </CardBody>
          <CardFooter className="text-muted">By {author}</CardFooter>
        </Card>
      </Col>
    )
  }
}

export default Article
