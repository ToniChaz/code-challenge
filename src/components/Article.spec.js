// Article.spec.js

import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import Article from './Article'

import { articles, article } from '../../test/mocks'

describe('Testing Article', () => {
  test('Renders correctly without articles', () => {
    const history = {
      push: jest.fn()
    }
    const tree = shallow(
      <Article history={history} />
    )
    expect(shallowToJson(tree)).toMatchSnapshot()
  })

  test('Renders correctly with articles', () => {
    const filterArticle = jest.fn()
    const match = { params: { articleId: '59eb8261d2ddc040ead6c7fa' } }
    const tree = shallow(
      <Article articles={articles} article={article} filterArticle={filterArticle} match={match} />
    )
    expect(shallowToJson(tree)).toMatchSnapshot()
  })
})
