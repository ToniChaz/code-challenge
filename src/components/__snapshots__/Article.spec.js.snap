exports[`Testing Article Renders correctly with articles 1`] = `
<Container
  tag="div">
  <Connect(Header) />
  <Row
    tag="div">
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }>
      <Connect(ArticleCard)
        article={
          Object {
            "_id": "59eb8261d2ddc040ead6c7fa",
            "author": "Bertram Wintheiser",
            "content": "Et iure enim saepe. Repellat reprehenderit sequi id. Eveniet sit ex. Rem et labore temporibus dicta architecto reprehenderit qui. Consequatur amet aspernatur porro. Quo consequatur nesciunt et quod id.",
            "excerpt": "Et iure enim saepe. Repellat reprehenderit sequi id. Eveniet sit ex. Rem et labore temporibus dicta ",
            "published": false,
            "tags": Array [
              "Avon portal Outdoors",
              "networks Berkshire Fish",
            ],
            "title": "Senior Communications Coordinator",
          }
        }
        single={true} />
    </Col>
  </Row>
  <Connect(Footer) />
</Container>
`;

exports[`Testing Article Renders correctly without articles 1`] = `""`;
